app/src/main/java/com/github/se/cyrcle/ui/list/ListScreen.kt

```diff
@@ -525,3 +616,134 @@ fun ActionCard(
+                .onGloballyPositioned { coordinates -> textFieldSize.value = coordinates.size },
+        colors = getOutlinedTextFieldColorsSearchBar(ColorLevel.PRIMARY),
+        trailingIcon = {
+          if (textFieldValue.value.isNotEmpty()) {
+            Icon(
+                imageVector = Icons.Filled.Clear,
+                contentDescription = "Clear search",
+                tint = defaultOnColor(),
+                modifier =
+                    Modifier.clickable {
+                      textFieldValue.value = ""
+                      showSuggestions.value = false
+                    })
+          }
+        },
+        singleLine = true,
+        keyboardOptions = KeyboardOptions.Default.copy(imeAction = ImeAction.Search),
+        keyboardActions =
+            KeyboardActions(
+                onSearch = {
+                  virtualKeyboardManager?.hide()
+
+                  runBlocking { addressViewModel.search(textFieldValue.value) }
+
+                  showSuggestions.value = true
+                }))
+
+    if (listOfSuggestions.value.size + 1 > 0) {
+      DropdownMenu(
+          expanded = showSuggestions.value,
+          onDismissRequest = { showSuggestions.value = false },
+          modifier =
+              Modifier.width(with(LocalDensity.current) { textFieldSize.value.width.toDp() })) {
+            if (permissionHandler.getLocalisationPerm().value) {
+              DropdownMenuItem(
+                  text = { Text("My Location") },
+                  onClick = {
+                    myLocation.value = true
+                    showSuggestions.value = false
+                    textFieldValue.value = "My Location"
+                    isTextFieldVisible.value = false
+                    focusManager.clearFocus()
+                  })
+            }
+
+            val seenNames = mutableSetOf<String>()
+            uniqueSuggestions.value =
+                listOfSuggestions.value.filter { suggestion ->
+                  val displayName =
+                      suggestion.suggestionFormatDisplayName(maxSuggestionDisplayNameLengthList)
+                  if (displayName in seenNames) {
+                    false
+                  } else {
+                    seenNames.add(displayName)
+                    true
+                  }
+                }
+
+            for (address in uniqueSuggestions.value.take(6)) {
+              DropdownMenuItem(
+                  text = {
+                    Text(address.suggestionFormatDisplayName(maxSuggestionDisplayNameLengthList))
+                  },
+                  onClick = {
+                    chosenLocation.value = address
+                    showSuggestions.value = false
+                    myLocation.value = false
+                    isTextFieldVisible.value = false
+                    textFieldValue.value =
+                        address.suggestionFormatDisplayName(maxSuggestionDisplayNameLengthList)
+                    focusManager.clearFocus()
```

Comment:
Just so I make sure I understand your whole code, what does this do and why is it here?

Reply:
Thanks for clarifying!